salt/utils/network.py

```diff
@@ -1906,3 +1906,59 @@ def dns_check(addr, port, safe=False, ipv6=None):
             raise SaltClientError()
         raise SaltSystemExit(code=42, msg=err)
     return resolved
+
+
+def parse_host_port(host_port):
+    '''
+    Takes a string argument specifying host or host:port.
+
+    Returns a (hostname, port) or (ip_address, port) tuple. If no port is given,
+    the second element of the returned tuple will be None.
+
+    host:port argument, for example, is accepted in the forms of:
+      - hostname
+      - hostname:1234
+      - hostname.domain.tld
+      - hostname.domain.tld:5678
+      - [1234::5]:5678
+      - 1234::5
+      - 10.11.12.13:4567
+      - 10.11.12.13
+    '''
+    # These are approximate regular expressions, intended only to distinguish
+    #   between the cases of host addressed as hostname, ipv4, or ipv6.
+    re_ipv4_port = re.compile('^(?P<host>\d{1,3}(?:\.\d{1,3}){3})(?:\:(?P<port>[0-9]{1,5}))?$')  # pylint: disable=W1401
+    re_ipv6_port = re.compile('^(?:\[(?P<host>(?:[0-9abcdef]{0,4}:)(?:(?:[0-9abcdef]{1,4})?:){1,6}[0-9abcdef]{0,4})\](?:\:(?P<port>[0-9]{1,5}))?|(?P=host))$')  # pylint: disable=W1401
+    re_hostname_port = re.compile('^(?P<host>(?:[a-zA-Z0-9]{1,62})(?:\.[a-zA-Z0-9]{1,62})*)(?:\:(?P<port>[0-9]{1,5}))?$')  # pylint: disable=W1401
```

Comment:
Regex strings written as `r''` in Python.

Reply:
#### A, this is Moot:
if you say regular expressions are undesirable, and this can't pass review without it, then why complain about string type of regular expression constant?

#### B, lint:
You are too slow to review PRs to be nitpicking stuff like this. I'm a volunteer and my employer is not paying me for my contributions to salt. Please decide if this is important enough to enforce consistently or not and implement it as a lint test (or at least submit a ticket so that it's in the backlog and someone else can do it).

Reply:
@aphor I pointed bugs in your code, pointed it is unnecessarily hard to maintain and mentioned it has no enough unit tests coverage. I also helped you with an example how to do it simpler. But I will not participate provocative discussions, sorry.